kv: add shared, replicated, and shared-replicated locks to TestEvaluateBatch

nvb opened the pull request:
Informs #91545.
Informs #100193.

This commit extends TestEvaluateBatch to include shared, replicated, and shared-replicated lock acquisition using Get, Scan, and ReverseScan requests.

Release note: None

nvb committed:
kv: add shared, replicated, and shared-replicated locks to TestEvaluateBatch

Informs #91545.
Informs #100193.

This commit extends TestEvaluateBatch to include shared, replicated, and
shared-replicated lock acquisition using Get, Scan, and ReverseScan requests.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111433)
<!-- Reviewable:end -->

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111433#-:-NfRtzY-CLx50uerEAAF:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/replica_evaluate_test.go` line 321 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111433#-NfRsVdEDHIbQvwwXRws:-NfRsVdEDHIbQvwwXRwt:bx8dh7n) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a6bebaaebc2f9bc9524899d7de5309d171bd9fc/pkg/kv/kvserver/replica_evaluate_test.go#L321)):*
> ```Go
> 			setup: func(t *testing.T, d *data) {
> 				writeABCDEFAt(t, d, ts.Prev())
> 				scanA := getArgsString("a")
> ```

nit: Should these say "getA" and "getG", here and elsewhere?

It's a bit weird because the verification method is called "verifyScanResult" as well, and then switches over `Get/Scan/ReverseScan` requests -- should we rename the thing to "verifyReadResult" instead? 
___
*[`pkg/kv/kvserver/replica_evaluate_test.go` line 384 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111433#-NfRt7wG7CpocfUl_VPA:-NfRt7wG7CpocfUl_VPB:btuz520) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a6bebaaebc2f9bc9524899d7de5309d171bd9fc/pkg/kv/kvserver/replica_evaluate_test.go#L384)):*
> ```Go
> 		{
> 			// Same as above, but with replicated shared locks.
> 			name: "gets with key locking (exclusive, replicated)",
> ```

"(shared, replicated)"


<!-- Sent from Reviewable.io -->

nvb committed:
kv: rename variables and a function in TestEvaluateBatch

This commit updates the naming of a few variables and the verifyScanResult
function to better reflect the use of GetRequests in the test.

Epic: None
Release note: None

nvb reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111433#-:-NfRymLv4vB_tr82hbn5:b-oduykh)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani)

___
*[`pkg/kv/kvserver/replica_evaluate_test.go` line 321 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111433#-NfRsVdEDHIbQvwwXRws:-NfRxmba1Mv0rHnlx1VY:b-o41r7n) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a6bebaaebc2f9bc9524899d7de5309d171bd9fc/pkg/kv/kvserver/replica_evaluate_test.go#L321)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: Should these say "getA" and "getG", here and elsewhere?

It's a bit weird because the verification method is called "verifyScanResult" as well, and then switches over `Get/Scan/ReverseScan` requests -- should we rename the thing to "verifyReadResult" instead?
</blockquote></details>

Done for both.
___
*[`pkg/kv/kvserver/replica_evaluate_test.go` line 384 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111433#-NfRt7wG7CpocfUl_VPA:-NfRyln_8xDMCWeQ8rRX:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a6bebaaebc2f9bc9524899d7de5309d171bd9fc/pkg/kv/kvserver/replica_evaluate_test.go#L384)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

"(shared, replicated)"
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
CI flakes:
> Cannot find commit 7c44d697313fb6cd4f4cd48c014c4943ed245c33 in the https://github.com/cockroachdb/cockroach repository

bors r+
